happy/HappyNodeJoin.py

```diff
@@ -231,6 +231,10 @@ def __add_new_interface_state(self):
         self.setNodeInterface(self.node_id, self.node_interface_name, new_node_interface)
 
     def __assign_network_addresses(self):
+        #  if node is a tap device, no need to assign IP to it.
+        if self.tap:
+            self.logger.error("HappyNodeJoin: {} is tap device, do not assign IP".format(self.node_id))
+            self.exit()
```

Comment:
Note to other reviewers: `self.exit()` will lead to a non-zero return from the process, as it ultimately calls `sys.exit(1)`